sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md

```diff
@@ -1,134 +1,147 @@
+    </p>
+    </details>
 
-    The Azure IoT SDK for C uses Eclipse Paho for C installed via vcpkg (for the cmake integration).
+### What is Covered
 
-    ```shell
-    $ cd
-    $ git clone https://github.com/Microsoft/vcpkg
-    $ cd vcpkg/
-    /vcpkg$ ./bootstrap-vcpkg.sh
-    /vcpkg$ ./vcpkg install paho-mqtt
-    ...
-    Total elapsed time: 2.835 min
+- Setup instructions for the Azure SDK for Embedded C suite.
+- Configuration, build, and run instructions for the IoT Hub Client Certificate Samples.
+
+_The following was run on an Ubuntu Desktop 18.04 environment, but it also works on WSL 1 and 2 (Windows Subsystem for Linux)._
 
-    The package paho-mqtt:x64-linux provides CMake targets:
+## Azure SDK for Embedded C Setup Instructions
 
-        find_package(eclipse-paho-mqtt-c CONFIG REQUIRED)
-        target_link_libraries(main PRIVATE eclipse-paho-mqtt-c::paho-mqtt3as-static eclipse-paho-mqtt-c::paho-mqtt3cs-static)
+1. Install Paho using vcpkg.
 
-    /vcpkg$
-    /vcpkg$ ./vcpkg integrate install
-    Applied user-wide integration for this vcpkg root.
+    The Azure IoT SDK for C uses Eclipse Paho for C installed via [vcpkg](https://github.com/Microsoft/vcpkg) (for the CMake integration).  This installation may take an extended amount of time.
 
-    CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake"
+    ```shell
+    git clone https://github.com/Microsoft/vcpkg
+    cd vcpkg/
+    ./bootstrap-vcpkg.sh
+    ./vcpkg install --triplet x64-linux curl cmocka paho-mqtt
     ```
 
-4. Clone the Azure SDK for C
+2.  Set the vcpkg environment variables.
+
+    Confirm `VCPKG_ROOT` is the path where vcpkg was cloned.
 
     ```shell
-    $ cd
-    $ git clone https://github.com/azure/azure-sdk-for-c
+    export VCPKG_DEFAULT_TRIPLET=x64-linux
+    export VCPKG_ROOT=PATH_TO_VCPKG #replace PATH_TO_VCPKG with full path to vcpkg/
     ```
 
-5. Generate a self-signed certificate
+    NOTE: Setting an environment variable only applies to the current session.  If you open a new window, the variable will have to be reset.
 
-    The Azure Embedded SDK for C IoT Client samples use a self-signed certificate.
-
-    A script is provided for creating that certificate.
+3. Clone the Azure Embedded SDK for C.
 
     ```shell
-    $ cd azure-sdk-for-c/sdk/samples/iot
-    azure-sdk-for-c/sdk/samples/iot$ ./generate_certificate.sh
+    cd ..
+    git clone https://github.com/azure/azure-sdk-for-c
+    ```
+
+## Configure and Run the IoT Hub Client Certificate Samples
+
+1. Generate a self-signed certificate.
+
+    **WARNING: This script is intended for sample use only and should not be used in any production-level code.**
+
+    The Azure Embedded SDK for C IoT Client Certificate Samples use a self-signed certificate. A script is provided for creating that certificate. You must use PowerShell Core to execute this script.
+
+    ```powershell
+    ./azure-sdk-for-c/sdk/samples/iot/generate_certificate.ps1
```

Comment:
on linux, I believe this can be run with
```bash
pwsh ./azure-sdk-for-c/sdk/samples/iot/generate_certificate.ps1
```

Reply:
hmmm `pwsh generate_certificate.pst1 works, but not `pwsh ./azure-sdk-for-c/sdk/samples/iot/generate_certificate.ps1`

Not sure why, but will leave as is for this merge.

Reply:
This is the given error to investigate. It looks like its just specific to the script being run, not a general issue.

```bash
momuno@DESKTOP-L0RIDFG:/mnt/c/Repos/momuno_azure-sdk-for-c/sdk/samples$ pwsh iot/generate_certificate.ps1
Can't open x509_config.cfg for reading, No such file or directory
139995857097152:error:02001002:system library:fopen:No such file or directory:../crypto/bio/bss_file.c:72:fopen('x509_config.cfg','r')
139995857097152:error:2006D080:BIO routines:BIO_new_file:no such file:../crypto/bio/bss_file.c:79:
unable to find 'distinguished_name' in config
problems making Certificate Request
139995857097152:error:0E06D06A:configuration file routines:NCONF_get_string:no conf or environment variable:../crypto/conf/conf_lib.c:270:
Exception: /mnt/c/Repos/momuno_azure-sdk-for-c/sdk/samples/iot/generate_certificate.ps1:31
Line |
  31 |      throw "'$Command' invocation failed with exit code ($LastExitCode …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | 'openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config
     | x509_config.cfg -subj "/CN=paho-sample-device1"' invocation failed with exit code (1)
```